app/models/setting.rb

```diff
@@ -7,6 +7,10 @@ def settings
     end
 
     def method_missing(method_name, *arguments, &block)
+      if settings.keys.exclude?(method_name.to_sym)
+        abort "ERROR: no setting found for #{method_name}. Please check settings in options.yml"
```

Comment:
Still working on getting the Rails.logger working, but maybe this is better than just logging? Should we abort when a setting is asked for that isn't listed as a key in the options.yml file? This would potentially fail during runtime at any point so it might disrupt users, but maybe we want to know sooner rather than later if we've written code against a setting that doesn't exist?

Reply:
I think failing is a good idea. The main thing I think we have to be careful of is adding a new setting which we expect to be there and it's a non-optional setting. That's the one situation where I think we'd break existing deployments, so we probably want to avoid that unless we really need to.

I was not aware of `abort` as a command.

Reply:
oh this should just check that they keys are in the options.yml file for any settings that we're using with `Setting.`. Since the options.yml file is checked in then it should always contain any keys that we're using with `Setting.` right?

Reply:
Oh, yes, that’s correct.